src/network/networkctl.c

```diff
@@ -268,12 +275,29 @@ static int list_links(int argc, char *argv[], void *userdata) {
 
                 t = link_get_type_string(links[i].iftype, d);
 
-                printf("%3i %-16s %-18s %s%-16s%s %s%-10s%s\n",
-                       links[i].ifindex, links[i].name, strna(t),
-                       on_color_operational, strna(operational_state), off_color_operational,
-                       on_color_setup, strna(setup_state), off_color_setup);
+                r = table_add_cell_full(table, NULL, TABLE_UINT32, (uint32_t *) &links[i].ifindex, (size_t) -1, (size_t) -1, 0, 100, 0);
```

Comment:
hmm, I think we should add TABLE_INT to the format-table.[ch] code, to make the cast unnecessary and this fully safe.